pkg/execution/executor/finalize.go

```diff
@@ -223,7 +223,7 @@ func (e *executor) finalizeEvents(ctx context.Context, opts execution.FinalizeOp
 			// Legacy - send inngest/function.failed, except for when the function has been cancelled.
 			if !strings.Contains(*opts.Response.Err, state.ErrFunctionCancelled.Error()) {
 				freshEvents = append(freshEvents, event.Event{
-					ID:        ulid.MustNew(uint64(now.UnixMilli()), rand.Reader).String(),
+					ID:        opts.Metadata.ID.RunID.String(),
```

Comment:
Could you add a comment explaining why we want the idempotency key to be the failing run ID?